src/main/java/co/cask/gcp/common/GCPUtils.java

```diff
@@ -48,8 +48,8 @@ public static ServiceAccountCredentials loadServiceAccountCredentials(String pat
     properties.put("fs.AbstractFileSystem.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS");
     String projectId = config.getProject();
     properties.put("fs.gs.project.id", projectId);
-    properties.put("fs.gs.system.bucket", GCSConfigHelper.getBucket(config.getPath()));
-    properties.put("fs.gs.working.dir", GCSConfigHelper.ROOT_DIR);
+    properties.put("fs.gs.system.bucket", GCSPath.from(config.getPath()).getBucket());
```

Comment:
`config.getPath()` may be null (as it is macro enabled) if `getFileSystemProperties` is called from prepareRun. https://github.com/data-integrations/google-cloud/blob/develop/src/main/java/co/cask/gcp/gcs/sink/GCSMultiBatchSink.java#L83

Reply:
right my bad!